flake.nix

```
@@ -25,18 +25,37 @@
           mesonBuildType = "release";
 
           outputs = [ "out" "dev" ];
-          outputBin = "dev";
 
           buildInputs = with pkgs; [ zlib ];
           nativeBuildInputs = with pkgs; [ pkg-config ninja meson ];
         };
         defaultPackage = packages.libspng;
 
-        # TODO checks
+        checks.libspng = pkgs.stdenv.mkDerivation {
+          pname = "libspng";
+          name = "libspng";
+          src = libspng-src;
+
+          doCheck = true;
+
+          mesonBuildType = "release";
+
+          mesonFlags = [
+            # this is required to enable testing
+            # https://github.com/randy408/libspng/blob/bc383951e9a6e04dbc0766f6737e873e0eedb40b/tests/README.md#testing
+            "-Ddev_build=true"
+          ];
+
+          outputs = [ "out" "dev" ];
+
+          checkInputs = with pkgs; [ cmake libpng ];
+          buildInputs = with pkgs; [ pkg-config zlib ];
+          nativeBuildInputs = with pkgs; [ ninja meson ];
+        };
```

I was thinking it might be a little cleaner to just override the `package`, since the derivation for running the tests is so similar:

```suggestion
        checks.libspng = packages.libspng.overrideAttrs (oldAttrs: {
          doCheck = true;

          mesonFlags = [
            # this is required to enable testing
            # https://github.com/randy408/libspng/blob/bc383951e9a6e04dbc0766f6737e873e0eedb40b/tests/README.md#testing
            "-Ddev_build=true"
          ];

          checkInputs = with pkgs; [ cmake libpng ];
        });
```

---

This is great. Thanks!